engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/UniqueItemsValidator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.constraintvalidators.hv;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.hibernate.validator.constraints.UniqueItems;
+
+/**
+ * Validates that the provided collection is a set of unique items
+ * <p>
+ * Uniqueness is defined by the equals method of the item being compared.
+ *
+ * @author Tadhg Pearson
+ */
+public class UniqueItemsValidator implements ConstraintValidator<UniqueItems, Collection<?>> {
+	@Override
+	public void initialize(UniqueItems uniqueItems) {
+		//No annotation fields to initialize
+	}
+
+	/**
+	 * @param objects Potentially null collection of input objects to check for duplicates
+	 * @param validatorContext Non-null {@link ConstraintValidatorContext} to accumulate validation messages
+	 *
+	 * @return True if the input collection is null or does not contain duplicate objects
+	 */
+	@Override
+	public boolean isValid(Collection<?> objects, ConstraintValidatorContext validatorContext) {
+		boolean valid = isValid( objects );
+		if ( !valid ) {
+			List<String> duplicates = findDuplicates( objects );
+			String message = "Contains the following duplicate items: " + duplicates;
+			validatorContext.disableDefaultConstraintViolation();
+			validatorContext.buildConstraintViolationWithTemplate( message ).addConstraintViolation();
```

Comment:
Instead of building a new message you won't be able to translate, you can use the default message and add parameters for the interpolation. You can find an example of that in `PatternValidator` (the unwrapping of the context and the addition of `regexp` as a message parameter).

Note that you should use `InterpolationHelper.escapeMessageParameter()` on the string to espace the parameter string.

Reply:
I'll take a look